core-cube/src/main/java/org/apache/kylin/cube/CubeManager.java

```diff
@@ -256,6 +256,23 @@ public CubeInstance createCube(CubeInstance cube, String projectName, String own
         }
     }
 
+    /**
+     * when clear all segments, it's supposed to reinitialize the CubeInstance
+     */
+    public CubeInstance clearSegments(CubeInstance cube) throws IOException {
+        try (AutoLock lock = cubeMapLock.lockForWrite()) {
+            cube = cube.latestCopyForWrite(); // get a latest copy
+            CubeUpdate update = new CubeUpdate(cube);
+            update.setToRemoveSegs(cube.getSegments().toArray(new CubeSegment[cube.getSegments().size()]));
+            update.setCuboids(Maps.<Long, Long> newHashMap());
+            update.setCuboidsRecommend(Sets.<Long> newHashSet());
+            update.setUpdateTableSnapshotPath(Maps.<String, String> newHashMap());
+            update.setCreateTimeUTC(System.currentTimeMillis());
```

Comment:
What's the reason to update the "createTimeUTC"?

Reply:
Now the mapping of CubeDesc & CubeInstance is one to one. It may not be necessary to have create time UTC for CubeInstance. Here, the create time UTC is regarded as the reinitialization time.